app/views/comments/_form.html.erb

```diff
@@ -65,50 +65,13 @@
 
   <%= javascript_include_tag "dragdrop" %>
   <script src="/lib/emoji.js" type="text/javascript"></script>
+  <script src="/assets/atwho_autocomplete.js" type="text/javascript"></script>
   <script>
     $('#comment-form').bind('ajax:beforeSend', function(event){
       $("#text-input").prop('disabled',true)
       $("#comment-form .btn-primary").button('loading',true);
     });
 
-    (function() {
-
-    var at_config = {
-      at: "@",
-      callbacks: {
-        remoteFilter: function(query, callback) {
-          $.getJSON("/api/srch/profiles?srchString=" + query, {}, function(data) {
-            if (data.hasOwnProperty('items') && data.items.length > 0) {
-              callback(data.items.map(function(i) { return i.docTitle }));
-            }
-           });
-          }
-        },
-        limit: 20
-      },
-      hashtags_config = {
-        at: "#",
-        callbacks: {
-          remoteFilter: function(query, callback) {
-            if (query != ''){
-              $.post('/tag/suggested/' + query, {}, function(response) {
-                 callback(response.map(function(tagnames){ return tagnames }));
-               });
-              }
-            }
-          },
-        limit: 20
-      },
-      emojis_config = {
-        at: ':',
-        data: Object.keys(emoji).map(function(name){ return {'name': name, 'value': emoji[name]} }),
-        displayTpl: "<li>${value} ${name}</li>",
-        insertTpl: ":${name}:", 
-        limit: 100
-     }
-
-    $('textarea#text-input').atwho(at_config).atwho(hashtags_config).atwho(emojis_config);
-
     $('#comment-form').bind('ajax:success', function(e, data, status, xhr){
```

Comment:
Can you confirm that the error message here still works without the `(function() {` around it? Maybe best to add that back in, as it helps the whole thing run /after/ page load is complete?

Reply:
I think adding it back in may be easier than testing it... i know you can get it to error either by turning off your locally running app OR by submitting a real unicode emoji (not just a `:smiley:`) -- since we're trying to fix that! But... could be a bit tricky to test.

Reply:
Hi @jywarren, I moved the code for error message back into the function athwho_autocomplete script! Quick question before I push this code --- so the public folder is in .gitignore-- how do we then make sure we're including /lib/emoji.js? I just realized it wasn't getting included when I made that code change above then freaked out momentarily because "emoji " wasn't defined then realized it was because the file is in .gitignore. Thanks again for all of your help!